docs/docs/development/hardware-integration/new-shield.mdx

```diff
@@ -197,190 +281,152 @@ this might look something like:
+Here is an example of a matrix transform for the previous example (18-key double macropad):
 
-```dts
-#include <dt-bindings/zmk/matrix_transform.h>
+```dts title="my_keyboard.dtsi"
+#include <dt-bindings/zmk/matrix_transform.h> // Put this with the other includes at the top of your dtsi
 
 / {
-    /* define kscan node with label `kscan0`... */
-
-    default_transform: keymap_transform_0 {
+    default_transform: keymap_transform0 {
         compatible = "zmk,matrix-transform";
-        columns = <8>;
-        rows = <8>;
-// | MX1  | MX2  | MX3  | MX4  | MX5  | MX6  | MX7  | MX8  | MX9  | MX10 | MX11 | MX12 | MX13 |    MX14     |
-// |   MX15   | MX16 | MX17 | MX18 | MX19 | MX20 | MX21 | MX22 | MX23 | MX34 | MX25 | MX26 | MX27 |  MX28   |
-// |    MX29    | MX30 | MX31 | MX32 | MX33 | MX34 | MX35 | MX36 | MX37 | MX38 | MX39 | MX40 |     MX41     |
-// |     MX42      | MX43 | MX44 | MX45 | MX46 | MX47 | MX48 | MX49 | MX50 | MX51 | MX52 |       MX53       |
-// |  MX54  |  MX55  |  MX56  |                  MX57                   |  MX58  |  MX59  |  MX60  |  MX61  |
+        columns = <6>;
+        rows = <3>;
         map = <
-RC(3,0)  RC(2,0) RC(1,0) RC(0,0) RC(1,1) RC(0,1) RC(0,2) RC(1,3) RC(0,3) RC(1,4) RC(0,4) RC(0,5) RC(1,6)     RC(1,7)
-RC(4,0)    RC(4,1) RC(3,1) RC(2,1) RC(2,2) RC(1,2) RC(2,3) RC(3,4) RC(2,4) RC(2,5) RC(1,5) RC(2,6) RC(2,7)   RC(3,7)
-RC(5,0)     RC(5,1) RC(5,2) RC(4,2) RC(3,2) RC(4,3) RC(3,3) RC(4,4) RC(4,5) RC(3,5) RC(4,6) RC(3,6)          RC(4,7)
-RC(6,0)       RC(6,1) RC(6,2) RC(6,3) RC(5,3) RC(6,4) RC(5,4) RC(6,5) RC(5,5) RC(6,6) RC(5,6)                RC(5,7)
-RC(7,0)    RC(7,1)   RC(7,2)                     RC(7,3)                    RC(7,5)    RC(7,6)    RC(6,7)    RC(7,7)
+        //  LKey 1 |LKey 2 |LKey 3      RKey 1 |RKey 2 |RKey 3
+            RC(0,0) RC(0,1) RC(0,2)     RC(0,3) RC(0,4) RC(0,5)
+        //  LKey 4 |LKey 5 |LKey 6      RKey 4 |RKey 5 |RKey 6
+            RC(1,0) RC(1,1) RC(1,2)     RC(1,3) RC(1,4) RC(1,5)
+        //  LKey 7 |LKey 8 |LKey 9      RKey 7 |RKey 8 |RKey 9
+            RC(2,0) RC(2,1) RC(2,2)     RC(2,3) RC(2,4) RC(2,5)
         >;
     };
+};
+```
+
+The above transform has 6 columns and three rows, while each half of the keyboard only has three columns and three rows.
+To allow the kscan matricies to be joined in the matrix transform, an offset is applied to the matrix transform of peripherals.
 
-    /* potentially other overlay nodes... */
+```dts title="my_keyboard_right.overlay"
+&default_transform { // Offset of 3 because the left side has 3 columns
+    col-offset = <3>;
 };
 ```
 
-Some important things to note:
+This offset means that when the right half of the keyboard has a keypress event triggered by the GPIO pins at the indicies `0,0` of its `row-gpios` and `col-gpios` arrays respectively, it will interpret it as an `RC(0,3)` event rather than an `RC(0,0)` event.
```

Comment:
```suggestion
This offset means that when the right half of the keyboard has a key event triggered by the GPIO pins at the indices `0,0` of its `row-gpios` and `col-gpios` arrays respectively, it will interpret it as an `RC(0,3)` event rather than an `RC(0,0)` event.
```